StripeFinancialConnections/StripeFinancialConnections/Source/Native/NativeFlowDataManager.swift

```diff
@@ -108,6 +111,10 @@ class NativeFlowAPIDataManager: NativeFlowDataManager {
         didUpdateManifest()
     }
 
+    func updateApiClient(with publishableKey: String) {
+        self.apiClient = STPAPIClient(publishableKey: publishableKey)
```

Comment:
is this safe? 🤔 

it feels like at this moment we have multiple different places where `apiClient` can be used and modified, and that sounds like it could create problems 

for example, it seems like here we execute extra functions on STPAPIClient:

https://github.com/stripe/stripe-ios/blob/e9b3c5bb/StripeFinancialConnections/StripeFinancialConnections/Source/FinancialConnectionsSheet.swift#L64-L68

Reply:
and does any SDK does this? ex. create multiple instances of `STPAPIClient`? basically a test of whether how necessary this is vs. trying to find some other solution

Reply:
Gonna revisit the approach I took here! Moving this PR to draft in the meantime